lib/internal/url.js

```diff
@@ -628,7 +629,7 @@ class URL {
 
   #onParseComplete = (href, origin, protocol, host, hostname, pathname,
                       search, username, password, port, hash, hasHost,
-                      hasOpaquePath) => {
+                      hasOpaquePath, hasHash) => {
```

Comment:
Why is `hasHash` necessary? Isn't it the same as just `hash !== ''`?

Reply:
Yes, you're right. I'm removing it.